js/frame.js

```diff
@@ -0,0 +1,83 @@
+function Frame(isFinal = false) {
+   this._firstRollScore = 0;
+   this._secondRollScore = 0;
+   this._thirdRollScore = 0;
+   this._rollCount = 0;
+   this._bonus = 0;
+   this._isFinalFrame = isFinal;
+};
+
+Frame.prototype = {
+   getTotalScore: function() {
+      return this._firstRollScore 
+               + this._secondRollScore 
+               + this._thirdRollScore 
+               + this._bonus;
+   },
+
+   isStrike: function() {
+      return this._firstRollScore === 10;
+   },
+
+   isSpare: function() {
+      return this._firstRollScore + this._secondRollScore === 10
+             && !this.isStrike();
+   },
+
+   isEnded: function() {
+      return this._isFinalFrame
```

Comment:
This method is difficult to understand. Can you make it more readable by extracting part of the logic, or using variables to give meaningful names?